eli5/keras/explain_prediction.py

```diff
@@ -85,54 +79,132 @@ def explain_prediction_keras(estimator, # type: Model
     :param layer:
         The activation layer in the model to perform Grad-CAM on:
         a valid keras layer name, layer index, or an instance of a Keras layer.
-        
-        If None, a suitable layer is attempted to be retrieved. 
-        The layer is searched for going backwards from the output layer, 
-        checking that the rank of the layer's output 
-        equals to the rank of the input.
 
+        If None, a suitable layer is attempted to be retrieved.
+        For best results, pick a layer that:
+
+        * has spatial or temporal information (conv, recurrent, pooling, embedding)
+          (not dense layers).
+        * shows high level features.
+        * has large enough dimensions for resizing over input to work.
 
-        :raises TypeError: if ``layer`` is not None, str, int, or keras.layers.Layer instance.
+
+        :raises TypeError: if ``layer`` is not None, str, int, or Kerass.layers.Layer instance.
         :raises ValueError: if suitable layer can not be found.
-        :raises ValueError: if differentiation fails with respect to retrieved ``layer``. 
+        :raises ValueError: if differentiation fails with respect to retrieved ``layer``.
     :type layer: int or str or keras.layers.Layer, optional
 
 
+    See :func:`eli5.explain_prediction` for more information about the ``model``,
+    ``doc``, and ``targets`` parameters.
+
+
+    Other arguments are passed to concrete implementations
+    for image and text explanations.
+
+
+    Returns
+    -------
+      expl : :class:`eli5.base.Explanation`
+        An :class:`eli5.base.Explanation` object for the relevant implementation.
+    """
+    # Note that this function should only do dispatch
+    # and no other processing
+    if image is not None:
+        return explain_prediction_keras_image(model,
+                                              doc,
+                                              image=image,
+                                              targets=targets,
+                                              layer=layer,
+                                              )
+    elif _maybe_image(model, doc):
```

Comment:
Not sure if it makes sense to warn in this case - I think if we say that ``image`` is optional, then we shouldn't raise warnings if user didn't pass it. If we remove the warning, then it's possible to join this with above "if" branch.